internal/app/api/handler/login/login.go

```diff
@@ -50,3 +65,101 @@ func (h *LoginHandler) GetUser(w http.ResponseWriter, r *http.Request) {
+	rawPassword := r.FormValue("password")
+	fullName := r.FormValue("fullname")
+	email := r.FormValue("email")
+	deliveryCenter := r.FormValue("deliverycenter")
+	dcs := []string{deliveryCenter}
+	bytes, err := bcrypt.GenerateFromPassword([]byte(rawPassword), 14)
+	if err != nil {
+		h.logger.Errorf("password encryption is fail err: %v", err)
+		respond.JSON(w, http.StatusUnauthorized, "register fail")
+		return
+	}
+	password := string(bytes)
+	user := types.User{
+		ID:             bson.NewObjectId(),
+		Username:       username,
+		FullName:       fullName,
+		Email:          email,
+		DeliveryCenter: dcs,
+		Password:       password,
+	}
+	err = h.srv.AddUser(&user)
+	if err != nil {
+		h.logger.Infof("can't save new user into database err: %v ", err)
+		respond.JSON(w, http.StatusInternalServerError, "add user fail")
+		return
+	}
+	respond.JSON(w, http.StatusCreated, "save user successfully")
+}
+func (h *LoginHandler) ChangeLocalPassword(w http.ResponseWriter, r *http.Request) {
+	username := r.FormValue("username")
+	password := r.FormValue("password")
+	newPassword := r.FormValue("newpassword")
+	_, err := h.loginSvr[1].Authenticate(username, password)
+	if err != nil {
+		h.logger.Errorf("user authentication is fail err:%v ", err)
+		respond.JSON(w, http.StatusUnauthorized, "you must type correctly your user name and password before you changes password")
+		return
+	}
+	user, err := h.srv.GetByName(username)
+	if err != nil {
+		h.logger.Errorf("can't get user form database  err %v", err)
+		respond.JSON(w, http.StatusInternalServerError, "can't get user form database")
+	}
+	hashPassword, err := bcrypt.GenerateFromPassword([]byte(newPassword), 14)
+	if err != nil {
+		h.logger.Errorf("encryption your password is fail err:%v", err)
+		respond.JSON(w, http.StatusInternalServerError, "have a error when update password")
+		return
+	}
+	user.Password = string(hashPassword)
+	h.logger.Infof("newPassword  : %v encrypt pass : %v", newPassword, string(hashPassword))
+	err = h.srv.ChangeUserPassword(user)
+	if err != nil {
+		h.logger.Errorf("can't update password err: %v,", err)
+		respond.JSON(w, http.StatusInternalServerError, "the password isn't change ")
+		return
+	}
+	respond.JSON(w, http.StatusOK, "your password been changed")
+	return
+}
+func (h *LoginHandler) SetLocalPassword(w http.ResponseWriter, r *http.Request) {
+	username := r.FormValue("username")
+	password := r.FormValue("password")
+	localPassword := r.FormValue("localpassword")
+	_, err := h.loginSvr[0].Authenticate(username, password)
+	if err != nil {
+		h.logger.Errorf("user ldap authentication is fail err:%v ", err)
+		respond.JSON(w, http.StatusUnauthorized, "you must type correctly your user name and password before change password")
+		return
+	}
+	user, err := h.srv.GetByName(username)
+	if err != nil {
+		h.logger.Infof("can't get user form database err:%v", err)
+		respond.JSON(w, http.StatusInternalServerError, "can't find user form database")
+		return
+	}
```

Comment:
This kind of thing is duplicated through all of your APIs. Please move to service layer...do it there and make sure you reuse it instead of duplicating things.